### PR TITLE
[statsdreceiver] add test for empty statsd tags

### LINKED ISSUE #32337: [statsdreceiver] fail to parse payloads with empty tag data
### Component(s)

receiver/statsd

## Description
sending a statsd payload w/ a completely empty tag causes a panic.

## Steps to Reproduce
send a payload like `test.metric:42|c|#,,,` - note the 0-length strings in between the commas denoting empty tags. dogstatsd clients sometimes are happy to send tags like this; i have mostly seen it in the go client.

## Expected Result
no panic, collector is calm and collected

## Actual Result
panic

### Collector version

v0.96.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by diurnalist**
**Description:** we see some payloads with completely empty tags, which we can safely ignore. This is already handled by the receiver, but we did not have a test for it.

Close #32337

**Testing:** adds simple test to receiver

**Documentation:** n/a

### CODE REVIEW

**Comment by diurnalist** [comment]
@jmacd @dmitryax I would appreciate a look at this one, I am trying to keep it up to date w/ the upstream changes. It does not look to me that the failed checks are related to the PR changes.

**Comment by diurnalist** [comment]
@atoulme is there any interest in this bugfix?

**Review by atoulme — APPROVED** [review]
looks simple enough, and guarding against input seems like a good idea.

**Comment by atoulme** [comment]
@diurnalist take a look at the conflict and see if your test passes with latest main without your modification.

**Comment by diurnalist** [comment]
@atoulme yeah I just noticed that a [change 4 days ago](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33683) obviated the need for this :p - i guess this PR's only contribution then is the test itself, I think the other PR fixed the panic issue as a side-effect :)

**Review by atoulme — APPROVED** [review]
More tests sound good thanks!

File: .chloggen/b_statsd-empty-tags.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [user]
```
Comment (dmitryax) [inline .chloggen/b_statsd-empty-tags.yaml:27]: We don't need a changelog for test. Please remove this

**Review by codeboten — APPROVED** [review]

**Pull request merged**